lib/pxu_auth0/auth0_api.ex

```diff
@@ -11,7 +11,11 @@ defmodule PxUAuth0.Auth0API do
 
   def process_request_body(body), do: Jason.encode!(body)
 
-  def process_response_body(body), do: body |> Jason.decode!()
+  def process_response_body(body) do
+    {_status, result} = Jason.decode(body)
```

Comment:
This might lead to some really strange results

do we catch Jason.DecodeError structs anywhere?

```iex(3)> Jason.decode('{"test": 1}')
{:ok, %{"test" => 1}}
iex(4)> Jason.decode('{"test":}')
{:error, %Jason.DecodeError{data: "{\"test\":}", position: 8, token: nil}}```

Reply:
we should just return the full tuple returned from decode probably

Reply:
Yeah, I was thinking it would bubble up to the `fetch_new_token` function in `AccessToken` but that's a little confusing. Doing a pattern match for that error is definitely more clear.